src/processing/server/board/BoardStateRequestHandler.java

```diff
@@ -0,0 +1,93 @@
+package processing.server.board;
+
+import java.io.IOException;
+
+import infrastructure.validation.logger.LogLevel;
+import infrastructure.validation.logger.ModuleID;
+import processing.utility.*;
+import networking.INotificationHandler;
+import processing.ClientBoardState;
+import processing.Serialize;
+
+/**
+ * This class implements INotificationHandler and handles
+ * Board State Requests from the clients. This handler will handle 
+ * all the messages with identifier BoardStateRequest
+ * 
+ * @author Himanshu Jain
+ * @reviewer Ahmed Zaheer Dadarkar
+ *
+ */
+
+public class BoardStateRequestHandler implements INotificationHandler{
+	
+	/**
+	 * Implementing the onMessageReceived function defined in
+	 * INotificationHandler interface.
+	 * 
+	 * @param message the message received from the networking module
+	 */
+	public void onMessageReceived(String message) {
+		
+		ClientBoardState.logger.log(
+				ModuleID.PROCESSING, 
+				LogLevel.INFO, 
+				"[#" + Thread.currentThread().getId() + "] "
+				+ "BoardState request is made by the client"
+		);
+		
+		String[] arrOfArguments = message.split(":", 2);
+		
+		// extract userID from the first part of the message
+		String user = arrOfArguments[1];
+		String[] id = user.split("_", 2);
+		UserId userId = new UserId(new IpAddress(id[0]), new Username(id[1]));
+		
+		/**
+		 * While requesting for BoardState the client passes it's address
+		 * with port as the argument.
+		 */
+		IpAddress clientAddress = new IpAddress(arrOfArguments[1]);
+		
```

Comment:
All this manipulation will become super annoying even if there's a requirement to add one extra field. Hence choosing JSON as a protocol will be much meaningful choice.